pages/admin/lessons/new/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react'
+import { gql, useLazyQuery } from '@apollo/client'
+import { AdminLessonCard } from '../../../../components/admin/lessons/AdminLessonCard'
+import { withGetApp, GetAppProps } from '../../../../graphql'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import Link from 'next/link'
+import styles from '../../../../scss/adminLessonPage.module.scss'
+
+const EXERCISES = gql`
+  query {
+    exercises {
+      flaggedAt
+      module {
+        lesson {
+          title
+        }
+      }
+    }
+  }
+`
+type Exercise = {
+  flaggedAt?: string
+  module: {
+    lesson: {
+      title: string
+    }
+  }
+}
+
+type Exercises = Exercise[]
+
+const Lessons: React.FC<GetAppProps> = ({ data }) => {
+  const { lessons } = data
+  const [lessonCards, setCards] = useState<[]>([])
+  const [getExercises, { data: exercisesData }] = useLazyQuery<{
+    exercises: Exercises
+  }>(EXERCISES, {
+    onCompleted: async () => {
+      const exerciseMapping = await exercisesData?.exercises.reduce(
+        (prev: any, cur) => {
+          if (cur.flaggedAt) {
+            if (!prev[cur.module?.lesson?.title]) {
+              prev[cur.module?.lesson?.title] = 1
+              return prev
+            }
+            prev[cur.module?.lesson?.title]++
+          }
+
+          return prev
+        },
+        {}
+      )
+
+      setCards(
```

Comment:
We should not be setting jsx in a react state. State should only contain the minimal amount of data needed to represent your UI's state. Use data from the state in order to perform correct logic for what you want to render.

React is based on the concept of a function rendering elements that are the result of its current values. If you store jsx in a state, thats an anti-react pattern since what you are rendering doesn't depend on current values.

Storing jsx in a state also creates the risk of having stale elements that don't correspond to the current set of properties; although the risk of `stale element` problem doesn't appear to happening in this PR, it is still an increased complexity. The increased complexity of `storing jsx in a state`doesn't offer any benefits over simply storing data in a state to be used in rendering logic. It just makes things more complex thus harder to maintain

Reply:
If an  array of objects is the minimum data needed to render `lessonsCardsComponents`(as per your example), then yeah that sounds right